mock-esignet-integration-impl/src/main/java/io/mosip/esignet/mock/integration/service/MockVCIssuancePlugin.java

```diff
@@ -0,0 +1,123 @@
+			Map<String, Object> identityDetails) {
+		JsonLDObject vcJsonLdObject = null;
+		try {
+			VCResult vcResult = new VCResult();
+			vcJsonLdObject = buildDummyJsonLDWithLDProof(holderId);
+			vcResult.setCredential(vcJsonLdObject);
+			return vcResult;
+		} catch (Exception e) {
+			log.error("Failed to build mock VC", e);
+			return null;
+		}
+	}
+
+private JsonLDObject buildDummyJsonLDWithLDProof(String holderId) throws IOException, GeneralSecurityException, JsonLDException, URISyntaxException {
+	  Map<String, Object> formattedMap = new HashMap<>();
+      formattedMap.put("id", holderId);
+      formattedMap.put("name", "John Doe");
+      formattedMap.put("age", 30);
+
+      Map<String, Object> verCredJsonObject = new HashMap<>();
+      verCredJsonObject.put("@context", "https://www.w3.org/2018/credentials/v1");
+      verCredJsonObject.put("type", Arrays.asList("VerifiableCredential"));
+      verCredJsonObject.put("id", holderId);
+      verCredJsonObject.put("issuer", "did:example:123");
+      verCredJsonObject.put("issuanceDate", getUTCDateTime());
+      verCredJsonObject.put("credentialSubject", formattedMap);
+
+      JsonLDObject vcJsonLdObject = JsonLDObject.fromJsonObject(verCredJsonObject);
+      vcJsonLdObject.setDocumentLoader(confDocumentLoader);
+      // vc proof
+      Date created = Date.from(LocalDateTime.parse((String)verCredJsonObject.get("issuanceDate"),
+                      DateTimeFormatter.ofPattern(UTC_DATETIME_PATTERN))
+              .atZone(ZoneId.systemDefault()).toInstant());
+      LdProof vcLdProof = LdProof.builder()
+              .defaultContexts(false)
+              .defaultTypes(false)
+              .type("RsaSignature2018")
+              .created(created)
+              .proofPurpose("assertionMethod")
+              .verificationMethod(discoveryKey)
+              .build();
+
+      URDNA2015Canonicalizer canonicalizer =	new URDNA2015Canonicalizer();
+      byte[] vcSignBytes = canonicalizer.canonicalize(vcLdProof, vcJsonLdObject);
+      String vcEncodedData = CryptoUtil.encodeToURLSafeBase64(vcSignBytes);
+
+      JWTSignatureRequestDto jwtSignatureRequestDto = new JWTSignatureRequestDto();
+      jwtSignatureRequestDto.setApplicationId(OIDC_SERVICE_APP_ID);
+      jwtSignatureRequestDto.setReferenceId("");
+      jwtSignatureRequestDto.setIncludePayload(false);
+      jwtSignatureRequestDto.setIncludeCertificate(true);
+      jwtSignatureRequestDto.setIncludeCertHash(true);
+      jwtSignatureRequestDto.setDataToSign(vcEncodedData);
+      JWTSignatureResponseDto responseDto = signatureService.jwtSign(jwtSignatureRequestDto);
+      LdProof ldProofWithJWS = LdProof.builder()
+              .base(vcLdProof)
+              .defaultContexts(false)
+              .jws(responseDto.getJwtSignedData())
+              .build();
+      ldProofWithJWS.addToJsonLDObject(vcJsonLdObject);
+      return vcJsonLdObject;	
+	}
+
+	public static String getUTCDateTime() {
```

Comment:
Make this private method, need not be static

Reply:
Done